internal/pkg/composable/providers/kubernetessecrets/config.go

```diff
@@ -10,4 +10,10 @@ import "github.com/elastic/elastic-agent-autodiscover/kubernetes"
 type Config struct {
 	KubeConfig        string                       `config:"kube_config"`
 	KubeClientOptions kubernetes.KubeClientOptions `config:"kube_client_options"`
+
+	TTL int `config:"ttl"`
+}
+
+func (c *Config) InitDefaults() {
+	c.TTL = 10
```

Comment:
I honestly don't know what should be best. But we were updating on bigger intervals than 1. I can change to 1 as default, and then the user can update it as they wish.

Reply:
Can you please check with 1min to see if we have less api calls?  (now rememberd saying 3-5-7 minutes)
And we can decide then

Reply:
When I checked, we were having around 24 calls at the same time because we were getting calls from every resource. Now to obtain the same number of calls, even with TTL set to 1, we would need 24 minutes.